.github/workflows/test.yml

```diff
@@ -29,3 +30,25 @@ jobs:
         run: cargo test --verbose
       - name: Run tests (rustls)
         run: cargo test --verbose --no-default-features --features=rustls
+  build-examples:
+    name: Build examples
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+      - name: Build examples
+        # TODO: build all examples with a single command?
+        run: |
+          cargo build --example chat_cli
```

Comment:
I've been using `cargo build --examples` locally, does that not work in GHA?

Reply:
Ha thanks, yes that does work. I wasn't aware that was a thing. I think I tried `cargo build --example --all` and then gave up.